src/pageboundary.cpp

```diff
@@ -97,7 +97,16 @@ int PageElementEnd::CastOffPages(FunctorParams *functorParams)
     assert(params);
 
     assert(params->m_currentPage);
-    this->MoveItselfTo(params->m_currentPage);
+
+    PageElementEnd *endBoundary = dynamic_cast<PageElementEnd *>(params->m_contentPage->Relinquish(this->GetIdx()));
```

Comment:
- Is it save here to assume that `this` is a child of `m_contentPage`? Otherwise this might result in a crash when calling `Relinquish` with arg -1. 

- I guess you could use `vrv_cast` for the cast.

Reply:
Apart from these minor comments, it looks good to me!

Reply:
>     * Is it save here to assume that `this` is a child of `m_contentPage`? Otherwise this might result in a crash when calling `Relinquish` with arg -1.

Yes, this is safe. But we can assert it.

>     * I guess you could use `vrv_cast` for the cast.

Yes, I can change this.

Reply:
> I guess you could use `vrv_cast` for the cast.

Actually that is one case where we do not want to use `vrv_cast` (i.,e. potentially a `static_cast`) because Object::Relinquish is not checking on the type for the object being remove. A `static_cast` on what is returned could be disastrous.